config/interpolate_funcs_test.go

```diff
@@ -2545,3 +2545,25 @@ func TestInterpolateFuncFlatten(t *testing.T) {
 		},
 	})
 }
+
+func TestInterpolateFuncURLEncode(t *testing.T) {
+	testFunction(t, testFunctionConfig{
+		Cases: []testFunctionCase{
+			{
+				`${urlencode("abc123-_")}`,
+				"abc123-_",
+				false,
+			},
+			{
+				`${urlencode("foo:bar@localhost?foo=bar&bar=baz")}`,
+				"foo%3Abar%40localhost%3Ffoo%3Dbar%26bar%3Dbaz",
+				false,
+			},
+			{
+				`${urlencode("mailto:email?subject=this+is+my+subject")}`,
+				"mailto%3Aemail%3Fsubject%3Dthis%2Bis%2Bmy%2Bsubject",
+				false,
+			},
```

Comment:
This set of test cases looks good! I'd like to suggest one more to cover the case of escaping something to place into a path segment:

* `foo/bar` should escape to `foo%2Fbar`

That way it should be safe to do things like, for example `"http://example.com/${urlencode(var.some_id)}/baz"` without worrying that `var.some_id` might contain slashes.